jsonapi/marshal.go

```diff
@@ -285,67 +287,68 @@ func getStructRelationships(relationer MarshalLinkedRelations, information Serve
 		relationships[name] = relationship
 	}
 
-	return &relationships
+	return relationships
 }
 
 // helper method to generate URL fields for `links`
 func getLinksForServerInformation(relationer MarshalLinkedRelations, name string, information ServerInformation) *Links {
-	links := &Links{}
-
-	if information != serverInformationNil {
-		prefix := strings.Trim(information.GetBaseURL(), "/")
-		namespace := strings.Trim(information.GetPrefix(), "/")
-		structType := getStructType(relationer)
+	if information == nil {
```

Comment:
Found some more info here: https://golang.org/doc/faq#nil_error.

I think so. I mean if the caller is funny and passes one of those weird nil typed interfaces, he should also be responsible for dealing with the panic. I'm not aware that it is a known go pattern to check all interface parameters to be "really" nil.